lib/spack/spack/test/cmd/load.py

```diff
@@ -136,3 +151,54 @@ def test_unload_fails_no_shell(install_mockery, mock_fetch, mock_archive,
 
     out = unload('mpileaks', fail_on_error=False)
     assert "To initialize spack's shell commands" in out
+
+
+def test_conflicting_load(database, working_env):
+    """Test that a compatible package is loaded if
+    one is available"""
+    # load the callpath that uses mpich
+    callpath_spec = spack.spec.Spec('callpath ^mpich').concretized()
+    mpileaks_spec = spack.spec.Spec('mpileaks ^mpich').concretized()
+    os.environ[uenv.spack_loaded_hashes_var] = callpath_spec.dag_hash()
+    # there are multiple mpileaks available, but only one that
+    # is compatible with our use of callpath ^mpich
+    out = load('--sh', 'mpileaks')
+    assert mpileaks_spec.dag_hash() in out
+
+
+def test_conflict_concrete(install_mockery, mock_fetch, mock_archive,
```

Comment:
This test needs to use the `tmpdir` fixture so that it will clean up after itself even on error.